protocol/binary_test.go

```diff
@@ -560,6 +561,66 @@ func TestBinaryEOFFailure(t *testing.T) {
 	}
 }
 
+type limitWriter struct {
+	b []byte
+}
+
+var errWriteLimitReached = errors.New("write limit reached")
+
+func (lw *limitWriter) Write(p []byte) (int, error) {
+	fmt.Printf("[limitWriter] write %v bytes\n", len(p))
```

Comment:
Is there a need for the print statement here? The table test description strings can capture the number of bytes written.